### ISSUE TITLE
Can't use link with Array

### ISSUE THREAD

**Issue by shaielc** [jupyter-widgets/traittypes#45]
Using link with Array for numpy array results in the exception:

`ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()`

```python
from traitlets import HasTraits, link
from traittypes import Array
import numpy as np

class Test(HasTraits):
    test = Array()
    test2 =Array()

test= Test()
link((test,"test"), (test,"test2"))
test.test =np.array([1,2,3])
```

**Comment by vidartf** [comment]
Hi! Could you please provide the full stack trace for you error as well ?

**Comment by shaielc** [comment]
> Hi! Could you please provide the full stack trace for you error as well ?

added the trace and pip list  ( thought it may help )

**Comment by vidartf** [comment]
You are right. This seems like it broke with traitlets 5, after this PR https://github.com/ipython/traitlets/pull/393. I think you could suggest a fix to traitlets where it uses a similar logic in `_update_target` as for the `set` method: https://github.com/ipython/traitlets/blob/0a8333904253a72964dd5b07837f5747f5c7c89b/traitlets/traitlets.py#L695-L700
